### PR TITLE
TRITON-2258 Want pkgsrc-tools installer in gz

### PR DESCRIPTION
**Pull request by bahamat**

### CODE REVIEW

**Review by danmcd — APPROVED** [review]
Is pkgsrc-setup.sh a new file?  Or an adopted-from-somewhere-else file?  Mostly curious, is all; it's not a blocker.  Looking forward to test results.

**Comment by bahamat** [comment]
It's new, but adapted from https://pkgsrc.joyent.com/

**Review by danmcd — APPROVED** [review]
Sorry I missed these during review 1.

**Review by danmcd — APPROVED** [review]
Damn, missed that too.

**Review by danmcd — APPROVED** [review]
Shoot.  I'd also take "/var/run" as an acceptable answer as well.

**Review by danmcd — DISMISSED** [review]

File: src/smartdc/bin/pkgsrc-setup.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+#
+# CDDL HEADER START
+#
```
Comment (danmcd) [inline src/smartdc/bin/pkgsrc-setup.sh:4]: Oh hell missed this earlier.  Use CDDL 1.0, not 1.1.  See piadm.sh for an example.

File: src/smartdc/bin/pkgsrc-setup.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+if [[ -n "$TRACE" ]]; then
+    export PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
+    set -o xtrace
+fi
+
+root="/${1}"
+
+# Occasionally, the filename and hash will need to be updated. Refer to
```
Comment (danmcd) [inline src/smartdc/bin/pkgsrc-setup.sh:37]: Thank you!

**Review by danmcd — COMMENTED** [review]
Shit, forgot the CDDL header you have should not be used, but the 1.0 one (more illumos friendly) should be used instead.  See piadm.sh for what you should copy/paste.

**Review by danmcd — APPROVED** [review]
Looking forward to test results.

**Comment by danmcd** [comment]
Please update the ticket with any of your own testing, @bahamat and I can IA this.

**Comment by danmcd** [comment]
Thank you for the additional test results.

**Pull request merged**